bot/cogs/webhook_remover.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import re
+
+from discord import Colour, Message
+from discord.ext.commands import Cog
+
+from bot.bot import Bot
+from bot.cogs.moderation.modlog import ModLog
+from bot.constants import Channels, Colours, Event, Icons
+
+WEBHOOK_URL_RE = re.compile(r"(discordapp\.com/api/webhooks/)(\d+/)(\S+/?)")
+
+ALERT_MESSAGE_TEMPLATE = (
+    "{user}, looks like you posted Discord Webhook URL to chat. "
+    "I removed this, but we **strongly** suggest to change this now "
+    "to prevent any spam abuse to channel. Please avoid doing this in future. "
+    "If you believe this was mistake, please let us know."
+)
+
+log = logging.getLogger(__name__)
+
+
+class WebhookRemover(Cog):
+    """Scan messages to detect Discord webhooks links."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @property
+    def mod_log(self) -> ModLog:
+        """Get current instance of `ModLog`."""
+        return self.bot.get_cog("ModLog")
+
+    async def delete_and_respond(self, msg: Message, url: str) -> None:
+        """Delete message and show warning when message contains Discord Webhook URL."""
+        # Don't log this, due internal delete, not by user. Will make different entry.
+        self.mod_log.ignore(Event.message_delete, msg.id)
+        await msg.delete()
+        await msg.channel.send(ALERT_MESSAGE_TEMPLATE.format(user=msg.author.mention))
+
+        message = (
+            f"{msg.author} (`{msg.author.id}`) posted Discord Webhook URL "
+            f"to #{msg.channel}. Webhook URL was `{url}`"
```

Comment:
`url` should be redacted in this method, or at the very least have a more descriptive variable name so we don't have to look into the `on_message` handler to see whether or not this method recieves the full URL or a redacted one.